### PR TITLE
booking ready by Husan

### PR DESCRIPTION
**Pull request by khusanKholmatov**

### CODE REVIEW

File: src/main/java/com/exadel/coolDesking/bot/util/MarkupService.java [hunk 1]
```diff
@@ -90,17 +99,99 @@ public InlineKeyboardMarkup listToKeyboard(List<String> data, String callbackDat
...
+    private InlineKeyboardMarkup getDays(int m, int year, String bookingBotId){
+        List<List<InlineKeyboardButton>> rowList = weekDays(m, year, bookingBotId);
+        LocalDate date = LocalDate.of(year, m, 1);
+        List<InlineKeyboardButton> buttonsRow = new ArrayList<>();
+        int k=1;
+        while(date.getMonthValue() == m) {
+            for (int i = 1; i <= 7; i++) {
+                if (i == date.getDayOfWeek().getValue() && date.getMonthValue() == m) {
+                    InlineKeyboardButton day = new InlineKeyboardButton();
+                    day.setText(k+"");
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/bot/util/MarkupService.java:116]: Integer.toString()

Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/bot/util/MarkupService.java:116]: you can use `date.getDayOfMonth()` instead of `k`

**Review by ohalushchak-exadel — CHANGES_REQUESTED** [review]

**Comment by ohalushchak-exadel** [comment]
@gargatuma please, take a look as well

File: src/main/java/com/exadel/coolDesking/bot/handler/oneDayBooking/OneDayBookingHandlerService.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        response.setUserState(UserState.SELECT_ONE_DAY);
+        return response;
+    }
+
+    protected void deleteCurrentBooking(UUID userId, String telegramId) {
+        incompleteBookingService.deleteByCreatedById(userId, telegramId);
+    }
+
+    private boolean isCorrectStartDate(LocalDate startDate) {
+        final int MAX_MONTH = 3;
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/bot/handler/oneDayBooking/OneDayBookingHandlerService.java:86]: this should be a constant field in the class

File: src/main/java/com/exadel/coolDesking/bot/handler/recurringBooking/RecurringDaysBookingHandlerService.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                markupService.parkingSelection(bookingBotId.toString())
+        );
+        response.setUserState(UserState.SELECT_PARKING);
+        return response;
+    }
+
+    protected BotResponseDto changeWeeks(CallbackQuery callbackQuery, int count, InlineKeyboardMarkup inlineKeyboardMarkup){
+        List<List<InlineKeyboardButton>> keyboard = changeMarkup(inlineKeyboardMarkup, count);
+        var response = new BotResponseDto();
+        final int MAX_WEEKS = 13;
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/bot/handler/recurringBooking/RecurringDaysBookingHandlerService.java:82]: this should be a constant field in the class

File: src/main/java/com/exadel/coolDesking/bot/util/MarkupService.java [hunk 4]
```diff
@@ -90,17 +101,99 @@ public InlineKeyboardMarkup listToKeyboard(List<String> data, String callbackDat
...
+                buttonsRow = new ArrayList<>();
+            }
+        }
+        rowList.add(buttonsRow);
+
+        return new InlineKeyboardMarkup(rowList);
+    }
+
+    private List<List<InlineKeyboardButton>> weekDays(int month, String bookingBotId){
+        List<String> days = new ArrayList<>(List.of("M", "T", "W", "T", "F", "S", "S"));
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/bot/util/MarkupService.java:178]: you don't need `new ArrayList<>`

File: src/main/java/com/exadel/coolDesking/bot/util/MarkupService.java [hunk 5]
```diff
@@ -90,17 +101,99 @@ public InlineKeyboardMarkup listToKeyboard(List<String> data, String callbackDat
...
+        int year = date.getYear();
+        int month = date.getMonthValue();
+        return getDays(month, year, bookingBotId);
+    }
+
+    private InlineKeyboardMarkup getDays(int month, int year, String bookingBotId){
+        List<List<InlineKeyboardButton>> rowList = weekDays(month, bookingBotId);
+        LocalDate date = LocalDate.of(year, month, 1);
+        List<InlineKeyboardButton> buttonsRow = new ArrayList<>();
+        int k=1;
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/bot/util/MarkupService.java:113]: it's not used anymore

**Review by ohalushchak-exadel — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/coolDesking/bot/dto/BookingDateIdDto.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import java.time.LocalDate;
+import java.util.UUID;
+
+@Setter
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+public class BookingDateIdDto {
+    private LocalDate date;
+    private UUID bookingBotId;
```
Comment (gargatuma) [inline src/main/java/com/exadel/coolDesking/bot/dto/BookingDateIdDto.java:17]: change name to incompleteBookingId

File: src/main/java/com/exadel/coolDesking/bot/UpdateReceiver.java [hunk 7]
```diff
@@ -8,7 +8,9 @@
 import com.exadel.coolDesking.user.UserState;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
 import org.telegram.telegrambots.meta.api.methods.BotApiMethod;
+import org.telegram.telegrambots.meta.api.methods.updatingmessages.DeleteMessage;
```
Comment (gargatuma) [inline src/main/java/com/exadel/coolDesking/bot/UpdateReceiver.java:13]: remove unused imports

**Review by gargatuma — CHANGES_REQUESTED** [review]

**Review by ohalushchak-exadel — APPROVED** [review]

**Review by gargatuma — APPROVED** [review]

**Pull request merged**